test/e2e/external_gateways.go

```diff
@@ -391,15 +358,211 @@ var _ = ginkgo.Describe("e2e multiple external gateway validation", func() {
+
+				tcpDumpSync := sync.WaitGroup{}
+				checkPingOnContainer := func(container string, wg *sync.WaitGroup) error {
+					defer ginkgo.GinkgoRecover()
+					defer tcpDumpSync.Done()
+					_, err := runCommand("docker", "exec", container, "timeout", "60", "tcpdump", "-c", "1", icmpCommand)
+					framework.ExpectNoError(err, "Failed to detect icmp messages on ", container, srcPingPodName)
+					framework.Logf("ICMP packet successfully detected on gateway %s", container)
+					return nil
+				}
+
+				tcpDumpSync.Add(2)
+				go checkPingOnContainer(gwContainer1, &tcpDumpSync)
+				go checkPingOnContainer(gwContainer2, &tcpDumpSync)
+
+				// Verify the external gateway loopback address running on the external container is reachable and
+				// that traffic from the source ping pod is proxied through the pod in the default namespace
+				ginkgo.By("Verifying connectivity via the gateway namespace to the remote addresses")
+
+				pingSync := sync.WaitGroup{}
+				// spawn a goroutine to asynchronously (to speed up the test)
+				// to ping the gateway loopbacks on both containers via ECMP.
+				for _, address := range addresses.targetIPs {
+					pingSync.Add(1)
+					go func(target string) {
+						defer ginkgo.GinkgoRecover()
+						defer pingSync.Done()
+						_, err := framework.RunKubectl(f.Namespace.Name, "exec", srcPingPodName, "--", "ping", "-c", testTimeout, target)
+						if err != nil {
+							framework.Logf("error generating a ping from the test pod %s: %v", srcPingPodName, err)
+						}
+					}(address)
+				}
+
+				pingSync.Wait()
+				tcpDumpSync.Wait()
+
+				ginkgo.By("Deleting one container")
+				deleteClusterExternalContainer(gwContainer2)
+				time.Sleep(3 * time.Second) // bfd timeout
+
+				tcpDumpSync = sync.WaitGroup{}
+				tcpDumpSync.Add(1)
+				go checkPingOnContainer(gwContainer1, &tcpDumpSync)
+
+				// Verify the external gateway loopback address running on the external container is reachable and
+				// that traffic from the source ping pod is proxied through the pod in the default namespace
+				ginkgo.By("Verifying connectivity via the gateway namespace to the remote addresses")
+				pingSync = sync.WaitGroup{}
+
+				for _, t := range addresses.targetIPs {
+					pingSync.Add(1)
+					go func(target string) {
+						defer ginkgo.GinkgoRecover()
+						defer pingSync.Done()
+						_, err := framework.RunKubectl(f.Namespace.Name, "exec", srcPingPodName, "--", "ping", "-c", testTimeout, target)
+						framework.ExpectNoError(err, "Failed to ping the remote gateway network from pod", target, srcPingPodName)
+					}(t)
+				}
+				pingSync.Wait()
+				tcpDumpSync.Wait()
+			},
+			ginkgotable.Entry("ipv4", &addressesv4, "icmp"),
+			ginkgotable.Entry("ipv6", &addressesv6, "icmp6"))
+
+		ginkgotable.DescribeTable("Should validate connectivity to an external gateway's loopback address via a pod with external gateway annotations enabled",
```

Comment:
this has the same title as the previous describe table, should you put for protocols TCP/UDP? and ICMP in the previous?

Reply:
Will do, yes